### PR TITLE
singleton init fixes

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Splitting these commits from #5929 since they are unrelated, and likely not affected by the PMI wire protocol refactoring effort.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/pmi

File: src/pmi/src/pmi_v1.c [hunk 1]
```diff
@@ -1215,7 +1215,9 @@ static int PMIi_InitIfSingleton(void)
 
     /* FIXME: We need to support a distinct kvsname for each
      * process group */
-    PMI_KVS_Put(singinit_kvsname, cached_singinit_key, cached_singinit_val);
+    if (cached_singinit_inuse) {
```
Comment (hzhou) [inline src/pmi/src/pmi_v1.c:1218]: Could you add a comment -- "only push the cached kvs if we have any"?  It is not obvious what `if (cached_singinit_inuse)` means.

**Review by hzhou — APPROVED** [review]
Approved with just a minor suggestion.

File: src/pmi/src/pmi_v1.c (hunk 1 above)
Comment (raffenet) [reply src/pmi/src/pmi_v1.c:1218]: done

**Pull request merged**